src/ddscxx/tests/SharedMemory.cpp

```diff
@@ -438,21 +448,161 @@ TYPED_TEST(SharedMemoryTest, invalid_shm_qos)
   dds::pub::qos::DataWriterQos w_qos;
   w_qos << dds::core::policy::Reliability::Reliable();
-  w_qos << dds::core::policy::Durability::TransientLocal();
+  w_qos << dds::core::policy::Durability::Transient();
   w_qos << dds::core::policy::History::KeepLast(10U);
   constexpr bool valid_w_shm_qos = false;
 
   EXPECT_NO_THROW(this->run_communication_test(DO_NOT_USE_ICEORYX, r_qos, w_qos, 10));
   EXPECT_NO_THROW(this->run_loan_support_api_test(valid_r_shm_qos, valid_w_shm_qos));
 }
 
+TYPED_TEST(SharedMemoryTest, writer_invalid_shm_qos)
+{
+  // valid reader shm QoS
+  dds::sub::qos::DataReaderQos r_qos;
+  r_qos << dds::core::policy::Reliability::Reliable();
+  r_qos << dds::core::policy::Durability::TransientLocal();
+  r_qos << dds::core::policy::History::KeepLast(10U);
+  constexpr bool valid_r_shm_qos = true;
+
+  // invalid writer SHM QoS
+  dds::pub::qos::DataWriterQos w_qos;
+  w_qos << dds::core::policy::Reliability::Reliable();
+  w_qos << dds::core::policy::Durability::Transient();
+  w_qos << dds::core::policy::History::KeepLast(10U);
+  constexpr bool valid_w_shm_qos = false;  
+
+  EXPECT_NO_THROW(this->run_communication_test(DO_NOT_USE_ICEORYX, r_qos, w_qos, 10));
+  EXPECT_NO_THROW(this->run_loan_support_api_test(valid_r_shm_qos, valid_w_shm_qos));
+}
+
+TYPED_TEST(SharedMemoryTest, reader_invalid_shm_qos)
+{
+  // invalid reader shm QoS
+  dds::sub::qos::DataReaderQos r_qos;
+  r_qos << dds::core::policy::Reliability::BestEffort();
+  r_qos << dds::core::policy::Durability::Transient();
+  r_qos << dds::core::policy::History::KeepLast(10U);
+
+  // valid writer SHM QoS
+  dds::pub::qos::DataWriterQos w_qos;
+  w_qos << dds::core::policy::Reliability::BestEffort();
+  w_qos << dds::core::policy::Durability::TransientLocal();
+  w_qos << dds::core::policy::History::KeepLast(10U);
+
+  EXPECT_NO_THROW(this->expect_no_communication(r_qos, w_qos));
+}
+
+TYPED_TEST(SharedMemoryTest, reliable_reader_does_not_match_best_effort_writer)
+{
+  // valid reader shm QoS
+  dds::sub::qos::DataReaderQos r_qos;
+  r_qos << dds::core::policy::Reliability::Reliable();
+  r_qos << dds::core::policy::Durability::TransientLocal();
+  r_qos << dds::core::policy::History::KeepLast(10U);
+
+  // valid writer SHM QoS
+  dds::pub::qos::DataWriterQos w_qos;
+  w_qos << dds::core::policy::Reliability::BestEffort();
+  w_qos << dds::core::policy::Durability::TransientLocal();
+  w_qos << dds::core::policy::History::KeepLast(10U);
+
+  this->expect_no_communication(r_qos, w_qos);
+}
+
+TYPED_TEST(SharedMemoryTest, best_effort_reader_receives_data_from_reliable_writer_via_shm)
+{
+  dds::sub::qos::DataReaderQos r_qos;
+  r_qos << dds::core::policy::Reliability::BestEffort();
+  r_qos << dds::core::policy::Durability::TransientLocal();
+  r_qos << dds::core::policy::History::KeepLast(10U);
+  constexpr bool valid_r_shm_qos = true;
+
+  // to set the durability service values of history depth
+  // TODO: except that this does not work, as the writer has no
```

Comment:
@sumanth-nirmal Either I use it wrong or it does not work (it compiles, but the value is not propagated as expected). This makes testing of https://github.com/eclipse-cyclonedds/cyclonedds/pull/1180/ harder.

Reply:
Needs to be understood/fixed before we consider merging this.

Reply:
Fixed and are now using DurabilityService QoS